sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/CbsTokenProvider.cs

```diff
@@ -68,7 +68,7 @@ internal sealed class CbsTokenProvider : ICbsTokenProvider
                                                   string appliesTo,
                                                   string[] requiredClaims)
         {
-            AccessToken token = await _credential.GetTokenAsync(new TokenRequest(requiredClaims), _cancellationToken);
+            AccessToken token = await _credential.GetTokenAsync(new TokenRequest(requiredClaims), _cancellationToken).ConfigureAwait(false);
```

Comment:
Good catch.   This was initially overlooked, but was recently fixed as part of ongoing efforts and will ride along with an forthcoming Event Hubs PR.   That said, I've no objection to it coming in via this PR as well.

Reply:
Rebased on top of latest changes